### PR TITLE
File cleanup

### PR DESCRIPTION
**Pull request by The-EDev**
- All scripts are now in `scripts/` directory
- Added QT Creator project files to gitignore
- `nginx_mime2cpp.py` can now run without `python`
- logo updates for documentation and README

### CODE REVIEW

File: scripts/nginx_mime2cpp.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python
```
Comment (mrozigor) [inline scripts/nginx_mime2cpp.py:1]: Maybe `python3` would be better?

File: CMakeLists.txt [hunk 2]
```diff
@@ -52,16 +52,16 @@ else()
 	add_test(NAME crow_test COMMAND ${CMAKE_CURRENT_BINARY_DIR}/tests/unittest)
 	add_test(NAME template_test COMMAND ${CMAKE_CURRENT_BINARY_DIR}/tests/template/test.py WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/tests/template)
 
-	file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/amalgamate)
+	file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/scripts)
 
-	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/amalgamate/crow_all.h
-		COMMAND python ${PROJECT_SOURCE_DIR}/amalgamate/merge_all.py
+	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/scripts/crow_all.h
```
Comment (mrozigor) [inline CMakeLists.txt:57]: Maybe just output resulting header to build dir?

Comment (The-EDev) [reply CMakeLists.txt:57]: right, will do

File: examples/example_with_all.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include "../amalgamate/crow_all.h"
+#include "../build/crow_all.h"
```
Comment (mrozigor) [inline examples/example_with_all.cpp:1]: `../crow_all.h`?

Comment (The-EDev) [reply examples/example_with_all.cpp:1]: `../crow_all.h` would imply copying crow_all to the root directory and using it from there. I can do that if you want, but it's not that necessary.

**Comment by The-EDev** [comment]
The tests are failing because of the body streaming test. The problem with that is that some data is being missed (which only happens with the test itself)

**Review by mrozigor — APPROVED** [review]

**Pull request merged**